pkg/vm/vm_test.go

```diff
@@ -538,6 +538,18 @@ func TestAPPENDStruct(t *testing.T) {
 	assert.Equal(t, &StructItem{[]StackItem{makeStackItem(5)}}, vm.estack.Pop().value)
 }
 
+func TestAPPENDCloneStruct(t *testing.T) {
+	prog := makeProgram(DUP, PUSH0, NEWSTRUCT, TOALTSTACK, DUPFROMALTSTACK, APPEND, FROMALTSTACK, PUSH1, APPEND)
+	vm := load(prog)
+	vm.estack.Push(&Element{value: &ArrayItem{}})
+	vm.Run()
+	assert.Equal(t, false, vm.state.HasFlag(faultState))
+	assert.Equal(t, 1, vm.estack.Len())
+	assert.Equal(t, &ArrayItem{[]StackItem{
+		&StructItem{[]StackItem{}},
```

Comment:
I'm not sure I understand why this is a correct result. My interpretation leaves me with three items on the stack, but I can't really prove it also, because tests have completely failed for this branch. Can you please rebase on top of current master which includes #353 fix?

Reply:
I've rebased.
Also `TO-`/`FROMALTSTACK` moves item between stacks, `APPEND` consumes both arguments without pushing anything. Why would it have 3 items?

Reply:
Yeah, probably I've misinterpreted `APPEND`, now I see that it's fine actually.